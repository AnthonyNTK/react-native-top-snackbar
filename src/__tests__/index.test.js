import { NativeModules } from 'react-native';

describe('TopSnackbar module', () => {
  let TopSnackbar;

  beforeAll(() => {
    NativeModules.SnackBar = {
      show: jest.fn(),
    };

    TopSnackbar = require('../index').default;
  });

  beforeEach(() => {
    NativeModules.SnackBar.show.mockClear();
  });

  it('calls native with default params', () => {
    TopSnackbar.show({});

    expect(NativeModules.SnackBar.show.mock.calls).toMatchSnapshot();
  });

  it('calls native with specified params', () => {
    TopSnackbar.show({
      message: 'Hello, World!',
      duration: NativeModules.SnackBar.SHORT,
      textColor: '#ffffff',
      backgroundColor: '#f4511e',
      leftIcon: 'example',
      iconPadding: 5,
      callback: () => alert('hihihi')
    });

    expect(NativeModules.SnackBar.show.mock.calls).toMatchSnapshot();
  });
});
