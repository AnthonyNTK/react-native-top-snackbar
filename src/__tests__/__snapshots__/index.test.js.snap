// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSnackbar module calls native with default params 1`] = `
Array [
  Array [
    "Empty String",
    undefined,
    "#ffffff",
    "#2aa2ef",
    "Done",
    "#ffffff",
    null,
    null,
    8,
    null,
  ],
]
`;

exports[`TopSnackbar module calls native with specified params 1`] = `
Array [
  Array [
    "Hello, World!",
    undefined,
    "#ffffff",
    "#f4511e",
    "Done",
    "#ffffff",
    "example",
    null,
    5,
    [Function],
  ],
]
`;
